client/consensus/common/src/import_queue.rs

```diff
@@ -117,12 +118,26 @@ pub trait ImportQueue<B: BlockT>: Send {
 		number: NumberFor<B>,
 		justifications: Justifications,
 	);
-	/// Polls for actions to perform on the network.
-	///
+}
+
+#[async_trait::async_trait]
+pub trait ImportQueue<B: BlockT>: Send {
+	/// Get a copy of the handle to [`ImportQueueService`].
+	fn service(&self) -> Box<dyn ImportQueueService<B>>;
+
+	/// Get a reference to the handle to [`ImportQueueService`].
+	fn service_ref(&mut self) -> &mut dyn ImportQueueService<B>;
+
 	/// This method should behave in a way similar to `Future::poll`. It can register the current
 	/// task and notify later when more actions are ready to be polled. To continue the comparison,
 	/// it is as if this method always returned `Poll::Pending`.
 	fn poll_actions(&mut self, cx: &mut futures::task::Context, link: &mut dyn Link<B>);
```

Comment:
DQ: Do we keep this around for compatibility reasons?

Reply:
Yes, there is still some work to be done for the import queue and I'd rather address those in a follow-up PR.